app/models/user.rb

```diff
@@ -67,6 +67,16 @@ def self.from_omniauth(access_token)
     user
   end
 
+  def self.from_oauth(oauth_user, provider)
+    User.create!(
+      name: oauth_user["name"],
+      email: oauth_user["email"],
```

Comment:
what if email is blank ?

Reply:
@tachyons `create!` will raise validation error which would be handled in the rescue block.. with appropriate error message..

```
rescue ActiveRecord::RecordInvalid
   api_error(status: 422, errors: @user.errors)
end
```

Reply:
@user is not assigned in this case, write a spec for the scenario where email is blank

Reply:
Oh is it.. will cover that case..

Reply:
Thanks @tachyons for this insight!! learned something new about how `bang!` only raises an exception.. Thanks :v:

Reply:
btw made the required changes and added specs.. please check @tachyons